extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ChunkInputStreamGenerator.java

```diff
@@ -3,181 +3,47 @@
-                            rowOffset, date -> {
-                                if (date == null) {
-                                    return QueryConstants.NULL_LONG;
-                                }
-                                final long epochDay = date.toEpochDay();
-                                if (epochDay < MIN_LOCAL_DATE_VALUE || epochDay > MAX_LOCAL_DATE_VALUE) {
-                                    throw new IllegalArgumentException("Date out of range: " + date + " (" + epochDay
-                                            + " not in [" + MIN_LOCAL_DATE_VALUE + ", " + MAX_LOCAL_DATE_VALUE + "])");
-                                }
-                                return epochDay * MS_PER_DAY;
-                            });
-                }
-                if (type == LocalTime.class) {
-                    return LongChunkInputStreamGenerator.<LocalTime>convertWithTransform(chunk.asObjectChunk(),
-                            rowOffset, time -> {
-                                if (time == null) {
-                                    return QueryConstants.NULL_LONG;
-                                }
-                                final long nanoOfDay = time.toNanoOfDay();
-                                if (nanoOfDay < 0) {
-                                    throw new IllegalArgumentException("Time out of range: " + time);
-                                }
-                                return nanoOfDay;
-                            });
-                }
-                // TODO (core#936): support column conversion modes
-
-                return new VarBinaryChunkInputStreamGenerator<>(chunk.asObjectChunk(), rowOffset,
-                        (out, item) -> out.write(item.toString().getBytes(Charsets.UTF_8)));
-            default:
-                throw new UnsupportedOperationException();
-        }
+    /**
+     * Creator of {@link ChunkInputStreamGenerator} instances.
+     * <p>
+     * This API may not be stable, while the JS API's usages of it are implemented.
+     */
+    interface Factory {
+        /**
+         * Returns an instance capable of writing the given chunk
+         *
+         * @param chunkType the type of the chunk to be written
+         * @param type the Java type of the column being written
+         * @param componentType the Java type of data in an array/vector, or null if irrelevant
+         * @param chunk the chunk that will be written out to an input stream
+         * @param rowOffset the offset into the chunk to start writing from
+         * @return an instance capable of serializing the given chunk
+         * @param <T> the type of data in the column
+         */
+        <T> ChunkInputStreamGenerator makeInputStreamGenerator(
+                final ChunkType chunkType,
+                final Class<T> type,
+                final Class<?> componentType,
+                final Chunk<Values> chunk,
```

Comment:
Why don't we read the chunk type from the chunk?

Future follow-up: use the ChunkReader.TypeInfo here instead of type/componentType/chunkType (and possible future need for arrow Field)?